src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs

```diff
@@ -102,6 +102,29 @@ public async Task<int> ExecuteCommand(PackageReferenceArgs packageReferenceArgs,
 
             var originalPackageSpec = matchingPackageSpecs.FirstOrDefault();
 
+            // Convert relative path to absolute path if there is any
+            List<string> sourcePaths;
+
+            if (packageReferenceArgs.Sources?.Any() == true)
+            {
+                sourcePaths = new List<string>();
+                string projectDirectory = Path.GetDirectoryName(packageReferenceArgs.ProjectPath);
+
+                foreach (string source in packageReferenceArgs.Sources)
+                {
+                    sourcePaths.Add(UriUtility.GetAbsolutePath(projectDirectory, source));
```

Comment:
I think this should be relative to the startup directory and not the project directory.

Try out how restore works, but I'm pretty confident that `dotnet restore --source` will work relative to the invocation directory.

Reply:
@nkolev92 
It looks current implementation is correct for now, because current stable dotnet (sdk 5.0.0 and 5.0.1) restore works with source relative to project not current invocation directory. Even "5.0.100-preview.8.20417.9" work relative to project location.
Please check below output:

```
dotnet restore C:\NuGetProj\IssueRepro\5127\App2\App2.csproj /p:restoreSources="..\Mysource2"
  Determining projects to restore...
C:\Program Files\dotnet\sdk\5.0.101\NuGet.targets(131,5): error : The local source 'C:\NuGetProj\IssueRepro\5127\Mysource2' doesn't exist. [C:\NuGetProj\IssueRepro\5127\App2\App2.csproj]
```

But starting from `5.0.200-preview.20614.14` it checks from relative to current invocation directory. Not sure if it's permanent change or now.
```
C:\Users\eryondon\Desktop\111\222> dotnet restore C:\NuGetProj\IssueRepro\5127\App2\App2.csproj /p:restoreSources="..\Mysource2"
  Determining projects to restore...
C:\Program Files\dotnet\sdk\5.0.200-preview.20614.14\NuGet.targets(131,5): error : The local source 'C:\Users\eryondon\Desktop\111\Mysource2' doesn't exist. [C:\NuGetProj\IssueRepro\5127\App2\App2.csproj]
```

Reply:
That's because of https://github.com/NuGet/Home/issues/9406 which was fixed in https://github.com/NuGet/NuGet.Client/pull/3722. 

5.0.100 has a bug, 5.0.200 preview has the fix. 
5.0.200 is the one with the correct behavior.